### PR TITLE
Replacing get_posts with wc_get_orders

### PR DESCRIPTION
**Pull request by wjrosa**
Fixes 2248-gh-woocommerce/woocommerce-subscriptions

## Description

This PR replaces the usage of the function `get_posts` with `wc_get_orders` to improve performance. To do that, I have updated all the parameters accordingly.

## How to test this PR

Code review. Check if the tests are still passing.

## Product impact

- [x] Added changelog entry (or does not apply)
- [x] Will this PR affect WooCommerce Subscriptions? yes
- [x] Will this PR affect WooCommerce Payments? yes
- [ ]  Added deprecated functions, hooks or classes to the [spreadsheet](https://docs.google.com/spreadsheets/d/1xw9xszcPMnWsp4C8OKZMsLzZob7tOmWT7qMqmEIq314/edit#gid=0)

### CODE REVIEW

**Review by mattallan — COMMENTED** [review]
Hey @wjrosa I'm sorry to say but these functions will no longer be used and will be deprecated once we merge https://github.com/Automattic/woocommerce-subscriptions-core/pull/163 so I don't think we need to update them 😢 Sorry!

I think we can also ignore the `get_posts()` used in `wcs_get_subscription_id_from_key()` as it's an old helper function from back when a subscription object used to be an array of data stored in order meta (stored in meta `{$order_id}_{$product_id}` which we referred to as a subscriptions key) 🙈 

I guess it technically still works and it's not a deprecated helper 🤷 So maybe it's fine for us to update it? Although, just above this `get_posts()` is a direct `posts` table query so we'd need to update that as well.

I'll leave it up to you if you want to update it or close this PR entirely.

If we do decide to leave it, we should probably add an inline comment in case someone stumbles upon this `get_posts()` and has questions xD

**Comment by wjrosa** [comment]
> Hey @wjrosa I'm sorry to say but these functions will no longer be used and will be deprecated once we merge #163 so I don't think we need to update them 😢 Sorry!
> 
> I think we can also ignore the `get_posts()` used in `wcs_get_subscription_id_from_key()` as it's an old helper function from back when a subscription object used to be an array of data stored in order meta (stored in meta `{$order_id}_{$product_id}` which we referred to as a subscriptions key) 🙈
> 
> I guess it technically still works and it's not a deprecated helper 🤷 So maybe it's fine for us to update it? Although, just above this `get_posts()` is a direct `posts` table query so we'd need to update that as well.
> 
> I'll leave it up to you if you want to update it or close this PR entirely.
> 
> If we do decide to leave it, we should probably add an inline comment in case someone stumbles upon this `get_posts()` and has questions xD

No worries! Thanks for letting me know. I have just added the comment above the `get_posts` call instead and removed the changelog entry 👍

**Review by mattallan — APPROVED** [review]
Thanks @wjrosa for the latest changes! Approving this :)

**Pull request merged**
